opennms-provision/opennms-provision-persistence/src/main/java/org/opennms/netmgt/provision/persist/requisition/RequisitionInterface.java

```diff
@@ -398,8 +400,12 @@ public void validate() throws ValidationException {
             throw new ValidationException("Requisition interface 'ip-addr' is a required attribute!");
         }
         if (m_monitoredServices != null) {
+            Set<String> serviceNameSet = new HashSet<>();
             for (final RequisitionMonitoredService svc : m_monitoredServices) {
                 svc.validate();
+                if(!serviceNameSet.add(svc.getServiceName())){
+                    throw new ValidationException("Duplicate service names are not allowed.");
```

Comment:
We may want to indicate what the duplicate service name we failed with was.

Reply:
True. That'll be a better user experience. Addressed.